### PR TITLE
request to merge assignment 3 into develop branch

### PR DESCRIPTION
**Pull request by munendra-successive**
4. Develop a data seeding module that generates mock API responses.

5. Create a sample POST API that returns a seeded data as a response.

6. Develop an authentication middleware using a JWT dummy token.

7. Integrate the authentication middleware with the previously created APIs.

8. Adhere to the specified directory structure.

9. Write a custom middleware function that logs the incoming requests' method, URL, and timestamp to the console.

10. Implement an error-handling middleware that captures errors thrown in the route handlers and sends an appropriate error response.

11.Write a series of middleware functions and chain them together to demonstrate how multiple middleware can be applied to a single route.

12.Build middleware that adds a custom header to every response. Allow the header value to be configurable.

13. Develop middleware that limits the number of requests a user can make in a given time frame. Include parameters for setting the limit.

### CODE REVIEW

File: src/MiddleWares/index.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import Logger from "./Logger.js";
```
Comment (UrvashiChaudhary) [inline src/MiddleWares/index.js:1]: Please fix the naming convention of the **Logger** import

File: src/MiddleWares/rateLimit.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import fs from "fs";
+
+function loadUserRequests() {
```
Comment (UrvashiChaudhary) [inline src/MiddleWares/rateLimit.js:3]: please use arrow functions

File: src/controllers/UserController.js
Comment (UrvashiChaudhary) [inline src/controllers/UserController.js:1]: We write controller functions name in camel case

File: src/controllers/UserController.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+const PostData = (req, res) => {
+  const data = req.body;
+  console.log(data);
+  if (data) {
+    res.send("Successfull");
+  }
+};
+
```
Comment (UrvashiChaudhary) [inline src/controllers/UserController.js:8]: Make the seeding module in utils folder and use it from there in the controllers to return the APIs responses

File: src/routes/otherRoutes.js
Comment (UrvashiChaudhary) [inline src/routes/otherRoutes.js:1]: Remove them if they are of no use

**Review by UrvashiChaudhary — CHANGES_REQUESTED** [review]

File: src/routes/otherRoutes.js
Comment (munendra-successive) [reply src/routes/otherRoutes.js:1]: Removed

File: src/controllers/UserController.js (hunk 3 above)
Comment (munendra-successive) [reply src/controllers/UserController.js:8]: Done

File: src/MiddleWares/rateLimit.js (hunk 2 above)
Comment (munendra-successive) [reply src/MiddleWares/rateLimit.js:3]: done

File: src/MiddleWares/index.js (hunk 1 above)
Comment (munendra-successive) [reply src/MiddleWares/index.js:1]: done

**Review by suraj-bha — APPROVED** [review]

**Pull request merged**